subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -315,31 +315,66 @@ export function updateRevenue(
 
   const deltaReserveBalance = reserveBalance.minus(assetStatus.reserveBalance);
 
-  const deltaProtocolSideRevenue = deltaReserveBalance
+  const newProtocolSideRevenue = deltaReserveBalance
     .toBigDecimal()
     .times(market.exchangeRate!) // convert to underlying
     .div(DECIMAL_PRECISION)
     .times(token.lastPriceUSD!);
 
-  // assume reserve fee is interest revenue
-  // revenue from liquidation reserve fee is handled in rollbackRevenue()
+  // AssetStatus.reserveBalance may include protocol side revenue
```

Comment:
Could you add an explanation of why we need to separate liquidations and repayments?

Reply:
fixed. Let me know what you think of the revised comments.